pkg/publish/ingress.go

```diff
@@ -176,74 +186,92 @@ func Ingress(req router.Request, svc *v1.ServiceInstance) (result []kclient.Obje
+					clusterDomainRules = append(clusterDomainRules, getIngressRule(svc, hostname, port.Port))
 				}
 			}
 		} else {
 			if len(ports) > 1 {
 				return nil, fmt.Errorf("multiple ports bound to the same hostname [%s]", hostname)
 			}
-			targets[hostname] = Target{Port: ports[0].TargetPort, Service: svc.Name}
-			rules = append(rules, getIngressRule(svc, hostname, ports[0].Port))
+			customDomainTargets[hostname] = Target{Port: ports[0].TargetPort, Service: svc.Name}
+			customDomainRules = append(customDomainRules, getIngressRule(svc, hostname, ports[0].Port))
 		}
 	}
 
-	if len(rules) == 0 {
-		return
-	}
+	for _, rules := range []struct {
+		rules  []networkingv1.IngressRule
+		name   string
+		target map[string]Target
+	}{
+		{rules: clusterDomainRules, name: clusterDomain, target: clusterDomainTargets},
+		{rules: customDomainRules, name: customDomain, target: customDomainTargets},
+	} {
+		if len(rules.rules) == 0 {
+			continue
+		}
+		// For custom domain, always use cert-manager to provision certificate.
+		useCertManager := rules.name == customDomain && cfg.CertManagerIssuer != nil && *cfg.CertManagerIssuer != ""
+		secrets, ingressTLS, err := setupCertsForRules(req, svc, rules.rules, useCertManager)
+		if err != nil {
+			return nil, err
+		}
 
-	secrets, ingressTLS, err := setupCertsForRules(req, svc, rules)
-	if err != nil {
-		return nil, err
-	}
+		targetJSON, err := json.Marshal(rules.target)
+		if err != nil {
+			return nil, err
+		}
 
-	targetJSON, err := json.Marshal(targets)
-	if err != nil {
-		return nil, err
-	}
+		proto := v1.PublishProtocolHTTP
+		if len(ingressTLS) > 0 {
+			proto = v1.PublishProtocolHTTPS
+		}
 
-	proto := v1.PublishProtocolHTTP
-	if len(ingressTLS) > 0 {
-		proto = v1.PublishProtocolHTTPS
-	}
+		hostnameSeen := map[string]struct{}{}
+		for _, rule := range rules.rules {
+			if _, ok := hostnameSeen[rule.Host]; ok {
+				continue
+			}
+			hostnameSeen[rule.Host] = struct{}{}
+			svc.Status.Endpoints = append(svc.Status.Endpoints, v1.Endpoint{
+				Address:         rule.Host,
+				PublishProtocol: proto,
+			})
+		}
 
-	hostnameSeen := map[string]struct{}{}
-	for _, rule := range rules {
-		if _, ok := hostnameSeen[rule.Host]; ok {
-			continue
+		ingress := &networkingv1.Ingress{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name.SafeConcatName(svc.Name, rules.name),
+				Namespace: svc.Namespace,
+				Labels:    svc.Spec.Labels,
+				Annotations: labels.Merge(svc.Spec.Annotations, map[string]string{
+					labels.AcornTargets: string(targetJSON),
+				}),
+			},
+			Spec: networkingv1.IngressSpec{
+				IngressClassName: ingressClassName,
+				Rules:            rules.rules,
+				TLS:              ingressTLS,
+			},
+		}
+		// if custom domains don't use cert-manager, use the default acorn cert-manager setting
+		if useCertManager && (ingress.Annotations["cert-manager.io/cluster-issuer"] == "" && ingress.Annotations["cert-manager.io/issuer"] == "") {
```

Comment:
Only setting cert-manager annotation if there is no cert-manager annotation being set... This is so that if user sets the annotation outside in acorn, we should respect that.